### PR TITLE
Add log action and prefix

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: lib/backend/model/rule.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ import (
 )
 
 type Rule struct {
-	Action string `json:"action,omitempty" validate:"backendaction"`
+	Action    string  `json:"action,omitempty" validate:"backendaction"`
+	LogPrefix *string `json:"log_prefix,omitempty" validate:"omitempty"`
```
Comment (caseydavenport) [inline lib/backend/model/rule.go]: Is there a mechanism in the code to return an error if LogPrefix is specified but the action is not "log"?

File: lib/api/rule.go [hunk 2]
```diff
@@ -24,7 +24,8 @@ import (
 )
 
 type Rule struct {
-	Action string `json:"action" validate:"action"`
+	Action    string  `json:"action" validate:"action"`
+	LogPrefix *string `json:"logPrefix,omitempty" validate:"omitempty"`
```
Comment (caseydavenport) [inline lib/api/rule.go]: If the action is "log", but the LogPrefix is nil, is there a default value, or just no prefix (null in etcd)?

**Comment by caseydavenport** [comment]
A couple of questions but otherwise LGTM.

I think we'll want that validation on the frontend API.  We shouldn't let people write config on that API that doesn't make sense.

File: lib/backend/model/rule.go (hunk 1 above)
Comment (robbrockbank) [inline lib/backend/model/rule.go]: The action does not need to be log.

**Comment by caseydavenport** [comment]
@robbrockbank - which milestone should this be placed in?

**Pull request closed without merging**